python/cudf/cudf/core/column/string.py

```diff
@@ -682,24 +698,20 @@ def sort_by_values(self, ascending=True, na_position="last"):
 
         idx_dev_arr = rmm.device_array(len(self), dtype="int32")
         dev_ptr = libcudf.cudf.get_ctype_ptr(idx_dev_arr)
```

Comment:
use empty column constructor and get ptr via data buffer

Reply:
Done